docker/domserver/scripts/start.d/50-domjudge.sh

```diff
@@ -89,17 +90,65 @@ else
 	echo "TRUSTED_PROXIES=${DOCKER_GATEWAY_IP}" >> webapp/.env.local
 fi
 
-# Add trusted proxies for Nginx
 NGINX_CONFIG_FILE=/etc/nginx/snippets/domjudge-inner
 
-# Remove the previous configuration 
+# Set up BaseURL
+
+# Fix BaseURL, Such as
+# "" -> "/"
+# "domjudge" -> "/domjudge"
+if [[ -z "${WEBAPP_BASEURL}" ]] || [[ "${WEBAPP_BASEURL:0:1}" != '/' ]]; then
+	WEBAPP_BASEURL="/${WEBAPP_BASEURL}"
+	echo "Fix WEBAPP_BASEURL ${WEBAPP_BASEURL:1} -> ${WEBAPP_BASEURL}"
+fi
+
+# Fix BaseURL, Such as
+# "/" -> "/"
+# "/domjudge/" -> "/domjudge"
+if [[ "${WEBAPP_BASEURL}" != "/" ]] && [[ "${WEBAPP_BASEURL: -1}" == '/' ]]; then
+	WEBAPP_BASEURL="${WEBAPP_BASEURL%?}"
+fi
+
+# Remove the previous location configuration
+sed -i "/^location \/.*/,/^\}/d" ${NGINX_CONFIG_FILE}
+
+if [[ "${WEBAPP_BASEURL}" == "/" ]]; then
```

Comment:
I assume this works if we run in multiple times? I.e. if restarting the image.

Flipping the env var would recreate the container right? So that is not a case we should handle

Reply:
But our container may restart multiple times, the code will be executed multiple times.

I deal with it this way, I can guarantee that the code is idempotent, right?

Reply:
If you tested it it sounds good

Reply:
I tested it locally and it looks okay.